### PR TITLE
Move broadcast hook to live-common

### PR DESCRIPTION
**Pull request by sprohaszka-ledger**
### 📝 Description

The `useBroadcast` hook is exactly the same between mobile and desktop, so it should be with other common hooks.

Unfortunately, in takes too much modification to enforce a stricter type (i.e. remove conditional value of `account?: AccountLike | null` from the hook).

### ❓ Context

- **Impacted projects**: `live-common`, `live-desktop` and `live-mobile`
- **Linked resource(s)**: [N/A] 

### ✅ Checklist

- [ ] **Test coverage** 
- [x] **Atomic delivery** 
- [x] **No breaking changes** 

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

### CODE REVIEW

File: apps/ledger-live-mobile/src/screens/Swap/Form/Modal/Confirmation.tsx [hunk 1]
```diff
@@ -17,14 +18,13 @@ import { addPendingOperation, getMainAccount } from "@ledgerhq/live-common/accou
...
 import { postSwapAccepted, postSwapCancelled } from "@ledgerhq/live-common/exchange/swap/index";
-import { getEnv } from "@ledgerhq/live-env";
 import { InstalledItem } from "@ledgerhq/live-common/apps/types";
+import { useBroadcast } from "@ledgerhq/live-common/hooks/useBroadcast";
 import { renderLoading } from "../../../../components/DeviceAction/rendering";
 import { updateAccountWithUpdater } from "../../../../actions/accounts";
 import DeviceAction from "../../../../components/DeviceAction";
 import QueuedDrawer from "../../../../components/QueuedDrawer";
 import ModalBottomAction from "../../../../components/ModalBottomAction";
-import { useBroadcast } from "../../../../components/useBroadcast";
```
Comment (gre) [inline apps/ledger-live-mobile/src/screens/Swap/Form/Modal/Confirmation.tsx:27]: does it need to be removed from the mobile project then?

Comment (sprohaszka-ledger) [reply apps/ledger-live-mobile/src/screens/Swap/Form/Modal/Confirmation.tsx:27]: Not sure to understand why. I just moved the import upper in the file and change the import path.
The `useBroadcast` file has been moved to live-common and the desktop one has been deleted.

File: libs/ledger-live-common/src/hooks/useBroadcast.ts [hunk 2]
```diff
@@ -15,7 +14,7 @@ type SignTransactionArgs = {
 export const useBroadcast = ({ account, parentAccount }: SignTransactionArgs) => {
   const broadcast = useCallback(
     async (signedOperation: SignedOperation): Promise<Operation> => {
-      invariant(account, "account not present");
+      if (!account) throw new Error("account not present");
```
Comment (chabroA) [inline libs/ledger-live-common/src/hooks/useBroadcast.ts:17]: Any specific reason not to keep using `invariant` here?

Also, nitpick but:
```suggestion
      if (!account) {
        throw new Error("account not present");
      }
```

Comment (sprohaszka-ledger) [reply libs/ledger-live-common/src/hooks/useBroadcast.ts:17]: Typescript is still complaining with `invariant` when we consider `account` as valid in the next lines.

Comment (gre) [reply libs/ledger-live-common/src/hooks/useBroadcast.ts:17]: 👍  yes. it's a good idea to get rid of `invariant` for this reason.

**Review by chabroA — APPROVED** [review]

**Review by haammar-ledger — COMMENTED** [review]
@sprohaszka-ledger Looks all good, just a changeset please? :slightly_smiling_face:

**Review by haammar-ledger — APPROVED** [review]
Thanks!

**Pull request merged**